src/components/InputBox.tsx

```diff
@@ -0,0 +1,188 @@
+  actionType?: 'none' | 'next' | 'done';
+  onAction?: () => void;
+  validator?: (text: string) => string | undefined;
+}
+
+export interface TextInputRef {
+  focus: () => void;
+  blur: () => void;
+  validate: () => boolean;
+}
+
+const InputBox: React.ForwardRefRenderFunction<TextInputRef, InputBoxProps> = (
+  {
+    leftIcon,
+    rightIcon,
+    onFocusColor,
+    value,
+    onChangeText,
+    placeholder,
+    keyboardType = 'default',
+    actionType = 'none',
+    onAction,
+    validator,
+    ...rest
+  },
+  ref
+) => {
+  const inputRef = React.createRef<TextInput>();
+  const [isFocused, setIsFocused] = useState(false);
+  const [error, setError] = useState<string | undefined>(undefined);
+  const leftIconColor = React.useRef<string>('#181F25');
+  const LeftIcon = leftIcon;
+
+  useImperativeHandle(ref, () => ({
+    focus: () => {
+      inputRef.current?.focus();
+    },
+    blur: () => {
+      inputRef.current?.blur();
+    },
+    validate: () => {
+      const errorMessage = validator ? validator(value) : undefined;
+      setError(errorMessage);
+      return !errorMessage;
+    },
+  }));
+
+  const handleFocus = () => {
+    leftIconColor.current = '#3E776D';
+    setIsFocused(true);
+    setError(undefined);
+  };
+
+  const handleBlur = () => {
+    setIsFocused(false);
+    const errorMessage = validator ? validator(value) : undefined;
+    setError(errorMessage);
+  };
+
+  const handleAction = () => {
+    if (validator) {
+      const isValid = validator(value);
+      if (isValid) {
+        onAction?.();
+      } else {
+        setError(isValid);
+      }
+    } else {
+      onAction?.();
+    }
+  };
+
+  return (
+    <View>
+      <View
+        style={[
+          styles.container,
+          isFocused ? styles.containerFocused : undefined,
+          error ? styles.errorOcurred : undefined,
+        ]}
+      >
+        {leftIcon && (
+          <View style={styles.leftIcon}>
+            <LeftIcon iconColor={error ? '#F2495C' : leftIconColor.current} />
+          </View>
+        )}
+        <TextInput
+          ref={inputRef}
+          style={[
+            styles.input,
+            isFocused && styles.inputFocused,
+            error && styles.inputError,
+          ]}
+          value={value}
+          onChangeText={onChangeText}
+          placeholder={placeholder}
+          keyboardType={keyboardType}
+          onFocus={handleFocus}
+          onBlur={handleBlur}
+          onSubmitEditing={handleAction}
+          returnKeyType={actionType === 'next' ? 'next' : 'done'}
+          {...rest}
+        />
+        {rightIcon && <Image source={rightIcon} style={styles.rightIcon} />}
+      </View>
+      {error && (
+        <View style={styles.errorContainer}>
+          <InfoIcon />
+          <TextLabel variant="errorLabel">{error}</TextLabel>
+        </View>
+      )}
+    </View>
+  );
+};
+
+const styles = StyleSheet.create({
```

Comment:
please create stylesheet file for custom components and put this stylesheet code in that file